src/tests/ftest/telemetry/telemetry_pool_metrics.py

```diff
@@ -158,8 +158,14 @@ def test_telemetry_pool_metrics(self):
             self.ior_cmd.dfs_chunk.update(self.ior_cmd.transfer_size.value)
             self.run_ior_with_pool(
                 timeout=200, create_pool=False, create_cont=False)
-        except TestFail:
-            self.log.info("#ior command failed!")
+        except TestFail as error:
+            # NOTE DAOS-12946:  it is not possible to properly test the metrics as we do not know
+            # how much data have been transferred.  However, the reason of the ior failure should be
+            # investigated and thus we fail the test.
+            self.log.error(
+                ">>>>>> Failure of IOR is unexpected and should be investigated: %s",
+                str(error))
+            self.fail("IOR command failed: {}".format(error))
 
```

Comment:
Sorry, I didn't realize before that this is `TestFail`. `self.fail()` will raise another `TestFail`. So we could just remove the `try..except` to get the same behavior. I think `Failure of IOR is unexpected and should be investigated` is implied. I.e. if IOR fails, the test fails and it's understood that it needs to be investigated.

Reply:
- [ ] Remove useless exception catch